python/pyfeelpp-mor/feelpp/mor/nirb/nirb.py

```diff
@@ -0,0 +1,1282 @@
+        self.Xh = tb.Xh
+        self.Dmu = tb.Dmu
+        self.Ndofs = tb.Ndofs
+
+        if self.worldcomm.isMasterRank():
+            print(f"[NIRB] Initialization done")
+            print(f"[NIRB] Number of nodes on the fine mesh : {self.tbFine.mesh().numGlobalPoints()}")
+
+    def getFieldSpace(self, coarse=False):
+        """Get the field space
+
+        Args:
+            coarse (bool, optional): get the coarse space. Defaults to False.
+
+        Returns:
+            feelpp._discr.*: field space
+        """
+        if not coarse:
+            if self.toolboxType == "heat":
+                return self.tbFine.spaceTemperature()
+            elif self.toolboxType == "fluid":
+                return self.tbFine.spaceVelocity()
+            else:
+                raise ValueError("Unknown toolbox")
+        else:
+            if self.toolboxType == "heat":
+                return self.tbCoarse.spaceTemperature()
+            elif self.toolboxType == "fluid":
+                return self.tbCoarse.spaceVelocity()
+            else:
+                raise ValueError("Unknown toolbox")
+
+
+    def initCoarseToolbox(self):
+        """Initialize the rectification problem
+        """
+        self.tbCoarse = self.setToolbox(self.H,mesh_path=self.coarsemesh_path)
+        self.XH = feelpp.functionSpace(mesh=self.tbCoarse.mesh(), order=self.order)
+        if self.worldcomm.isMasterRank():
+            print(f"[NIRB] Number of nodes on the coarse mesh : {self.tbCoarse.mesh().numGlobalPoints()}")
+
+
+    def setToolbox(self, hsize, mesh_path=None):
+        """Set up the toolbox object for the given model and mesh
+
+        Args:
+        -----
+            hsize (float): size of the mesh
+
+        Returns:
+        --------
+            Toolbox: toolbox object
+        """
+        if mesh_path == None: mesh_path=self.finemesh_path
+        # load meshes
+        mesh_ = feelpp.mesh(dim=self.dimension, realdim=self.dimension)
+        mesh = feelpp.load(mesh_, mesh_path, hsize)
+
+        # set mesh and model properties
+        if self.toolboxType == "heat":
+            tb = heat.heat(dim=self.dimension, order=self.order)
+        elif self.toolboxType == "fluid":
+            tb = fluid.fluid(dim=self.dimension)
+        else:
+            raise ValueError("Unknown toolbox")
+
+        tb.setMesh(mesh)
+        tb.setModelProperties(self.model)
+
+        tb.init()
+
+        return tb
+
+    def getField(self, toolbox):
+        """Get field of interest from the toolbox
+
+        Args:
+            toolbox (Toolbox): Tolbox object
+
+        Raises:
+            ValueError: Unknow toolbox
+
+        Returns:
+            feelpp_.discr.Element_*: field of the solution
+        """
+        if self.toolboxType == "heat":
+            return toolbox.fieldTemperature()
+        elif self.toolboxType == "fluid":
+            return toolbox.fieldVelocity()
+        else:
+            raise ValueError("Unknown toolbox")
```

Comment:
how about using a dict to map the function calls ? that way it extends in a nicer way to other toolboxes